core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/rewrites/ReplaceShapelessTaggedDecimal.scala

```diff
@@ -27,15 +31,57 @@ class ReplaceShapelessTaggedDecimal extends SyntacticRule("ReplaceShapelessTagge
 
       case t @ Type.Apply(
         t"@@",
-        t"scala.math.BigDecimal" :: Type.Tuple(Type.Select(_, precisionNat) :: Type.Select(_, scaleNat) :: Nil) :: Nil
+        List(
+          t"scala.math.BigDecimal",
+          Type.Tuple(
+            List(
+              Type.Select(_, precisionNat),
+              Type.Select(_, scaleNat)
+            )
+          )
+        )
       ) =>
-        val precision = precisionNat.value.stripPrefix("_")
-        val scale = scaleNat.value.stripPrefix("_")
-        val taggedDecimalType = s"_root_.higherkindness.mu.rpc.avro.Decimals.TaggedDecimal[$precision, $scale]"
-        Patch.replaceTree(t, taggedDecimalType)
+        val precision = precisionNat.value.stripPrefix("_").toInt
+        val scale = scaleNat.value.stripPrefix("_").toInt
+        Patch.replaceTree(t, taggedDecimalType(precision, scale))
+
+      case t @ Type.Apply(
+        t"@@",
+        List(
+          t"scala.math.BigDecimal",
+          Type.Tuple(
+            List(
+              Type.Tuple(Type.Select(_, precisionNat1) :: Type.Select(_, precisionNat2) :: Nil),
+              Type.Select(_, scaleNat)
+            )
+          )
+        )
+      ) =>
+        val precision = precisionNat1.value.stripPrefix("_").toInt * 10 + precisionNat2.value.stripPrefix("_").toInt
+        val scale = scaleNat.value.stripPrefix("_").toInt
+        Patch.replaceTree(t, taggedDecimalType(precision, scale))
+
+      case t @ Type.Apply(
+        t"@@",
+        List(
+          t"scala.math.BigDecimal",
+          Type.Tuple(
+            List(
+              Type.Tuple(Type.Select(_, precisionNat1) :: Type.Select(_, precisionNat2) :: Nil),
+              Type.Tuple(Type.Select(_, scaleNat1) :: Type.Select(_, scaleNat2) :: Nil)
```

Comment:
We could have a scale of two digits and a precision of one. Not sure if we need to consider this here in this pattern matching. I can't remember the place, but we were like making a validation around that in the refactor.

```bash
scala> val bd = BigDecimal("0.000000000012345")
val bd: scala.math.BigDecimal = 1.2345E-11

scala> bd.scale
val res4: Int = 15

scala> bd.precision
val res5: Int = 5
```

Reply:
Awesome, thanks for the explanation